lib/shared/models/WeightEntry.dart

```diff
@@ -1,6 +1,27 @@
 class WeightEntry {
+  final String? id;
   final DateTime timestamp;
   final double weight;
 
-  WeightEntry({required this.timestamp, required this.weight});
+  WeightEntry({
+    this.id,
+    required this.timestamp,
+    required this.weight,
+  });
+
+  Map<String, dynamic> toFirestore() {
+    return {
+      'id': id,
+      'timestamp': timestamp.toIso8601String(),
```

Comment:
use Firestore Timestamps instead when storing the datetime as they are better for filtering etc